contracts/sol6/KyberStorage.sol

```diff
@@ -81,39 +100,22 @@ contract KyberStorage is IKyberStorage, PermissionGroupsNoModifiers, Utils5 {
         emit ReserveRebateWalletSet(reserveId, rebateWallet);
     }
 
-    function setContracts(IKyberFeeHandler _feeHandler, address _matchingEngine)
+    function setContracts(address _feeHandler, address _matchingEngine)
         external
         override
     {
         onlyNetwork();
-        require(_feeHandler != IKyberFeeHandler(0), "feeHandler 0");
+        require(_feeHandler != address(0), "feeHandler 0");
         require(_matchingEngine != address(0), "matchingEngine 0");
-        IKyberMatchingEngine newMatchingEngine = IKyberMatchingEngine(_matchingEngine);
 
-        if (feeHandler.length > 0) {
-            feeHandler.push(feeHandler[0]);
-            feeHandler[0] = _feeHandler;
-        } else {
-            feeHandler.push(_feeHandler);
-        }
-
-        if (matchingEngine.length > 0) {
-            matchingEngine.push(matchingEngine[0]);
-            matchingEngine[0] = newMatchingEngine;
-        } else {
-            matchingEngine.push(newMatchingEngine);
-        }
+        feeHandlerHistory.saveContract(_feeHandler);
```

Comment:
if history contract isn't set will revert in a non elegant way.
this is what u meant?

Its not that bad.
but we might want to issue some uninitialized revert message.
maybe both can have the same message.